DoubleSharp/Concurrency/ConcurrencyExtensions.cs

```diff
@@ -1,44 +1,166 @@
-namespace DoubleSharp.Concurrency; 
+using System.Runtime.CompilerServices;
 
+namespace DoubleSharp.Concurrency;
+
+/// <summary>DoubleSharp concurrency extensions.</summary>
 public static class ConcurrencyExtensions {
-	public static T Locked<T>(this T @object, Action functor) {
-		if(@object == null) throw new ArgumentNullException(nameof(@object));
-		lock(@object)
-			functor();
+
+	/// <summary>
+	/// Locks the target object and performs the specified action.
+	/// </summary>
+	/// <typeparam name="T">The type of the object.</typeparam>
+	/// <param name="object">The object.</param>
+	/// <param name="action">The <see cref="Action"/> delegate to perform.</param>
+	/// <returns>The object.</returns>
+	/// <exception cref="ArgumentNullException"><paramref name="object"/> is null.</exception>
+	public static T Locked<T>(this T @object, Action action) {
+		if (@object == null) throw new ArgumentNullException(nameof(@object));
+		lock (@object)
+			action();
 		return @object;
 	}
-	
-	public static T Locked<T>(this T @object, Action<T> functor) {
-		if(@object == null) throw new ArgumentNullException(nameof(@object));
-		lock(@object)
-			functor(@object);
+
+	/// <summary>
+	/// Locks the target object and performs the specified action on it.
```

Comment:
We might want to make note that this variant passes the object along to the functor. Not a blocker for merging, just future improvement to consider.

Reply:
The "on it" at the end was intended to distinguish that, but there might be better language for that.